### ISSUE TITLE
similarity calculation for sets of phenotypes

### ISSUE THREAD

**Issue by hannahblau** [pnrobinson/phenoCompare#22]
What similarity function to implement for comparing sets of HPO terms in the gsppc (gene set phenotypic profile comparison) R notebook? The first version used Jaccard similarity. Could re-implement one of the similarity functions from
HPOSim: An R Package for Phenotypic Similarity Measure and Enrichment Analysis Based on the Human Phenotype Ontology
Yue Deng, Lin Gao, Bingbo Wang, Xingli Guo
PLOS ONE, 9 February 2015

However, I don't think this paper has a good metric for comparing sets qua sets. They propose computing the pairwise similarity of every pair of elements (one from the first set, one from the second set) and then choosing the max similarity or the mean or...

**Comment by pnrobinson** [comment]
After looking at the new annotations, I think that we could do the following

1. Define the target set as the union of all of the terms in the GPI disease definitions (i.e., the main HPO files).
2. Take the information content of each of the terms based on their frequency in the entire HPO database
3. For each of the terms in the GPI-target-protein diseases, find the best match (most informative common ancestor) in the first set. Record its information content, and take the sum of the ICs for all terms. Then, take the sum of the ICs for all of the  GPI-target-protein diseases.
4. For the randomization, we could either define the target set randomly (i.e., use random diseases instead of GPI diseases) or the other set randomly.
5. This way of comparing the sets will take the specifity of the matches into account much more than the Jaccard approach.

**Comment by LCCarmody** [comment]
Random interjection, but I think this may have an effect on the results.  Is there an argument to remove certain annotations for comparison?  Specifically, I have noticed that "Autosomal recessive" or "Autosomal dominant" or "early onset" are listed as annotations.  While this is important information for the disease, is it reasonable to use these as phenotypes?

**Comment by hannahblau** [comment]
I noticed those "phenotypes" myself but did not know what to do with them. Currently there is no filtering on the phenotypes, they are drawn from the annotations files available on [http://compbio.charite.de/jenkins/job/hpo.annotations.monthly/lastSuccessfulBuild/artifact/annotation/](http://compbio.charite.de/jenkins/job/hpo.annotations.monthly/lastSuccessfulBuild/artifact/annotation/) and I did not think I should second-guess what's listed as an annotation even though some of them are really modes of inheritance. I could filter these out if I locate the common ancestor of the modes of inheritance and then I could toss out any child of that HPO term. It would be an extra step but I think not a difficult one.

**Comment by LCCarmody** [comment]
Depending on if Peter agrees, I think it would be a good idea to filter out any annotations that are not under "Phenotypic Abnormality".  I think that would eliminate all of these annotations.  @pnrobinson What do you think?

**Comment by pnrobinson** [comment]
Leigh's idea is good, we should just use terms under "Phenotypic Abnormality".

**Comment by hannahblau** [comment]
per discussion with @pnrobinson :
format 
col1=OMIM id of a disease
col2=EntrezGene id of the corresponding disease gene (there may be multiple entries for one 
disease(
col3=shows whether it is a disease ("phenotype") or elsewise (e.g., gene). We are just interested in the phenotype rows:

103780  125     phenotype 

medgen ncbi website
private final static String MIM2GENE_MEDGEN_URL = "ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/mim2gene_medgen";

parse the lists of GPI and GPI target genes and use mim2gene_medgen file to get the corresponding disease ids.
Use OMIM website to check correctness

1. Get the phenotype.hpoa from here
http://compbio.charite.de/jenkins/job/hpo.annotations.2018/
2. Get the latest hp.obo from here
https://hpo.jax.org/app/download/ontology​

**Comment by hannahblau** [comment]
@pnrobinson @LCCarmody 
Three genes from the GPI pathway/anchored groups have associated OMIM entries listed in the _mim2gene_medgen_ file, but the OMIM ids do not appear in _phenotype.hpoa_. They are:

Gene                | Group |Disease
-------------- | ------ | ---------
PIGS (ENTREZ:94005) | Pathway | OMIM:618143
ART4 (ENTREZ:420) | Anchored | OMIM:616060
SEMA7A (ENTREZ:8482) | Anchored | OMIM:614745

Here are the corresponding lines in the _mim2gene_medgen_ file:
```
618143            94005  phenotype       GeneMap       NULL    -
616060            420      phenotype       GeneMap       C1292294        nondisease
614745            8482    phenotype       GeneMap       C3553633        nondisease
``` 
Decided to ignore any line of _mim2gene_medgen_ that contains "nondisease" in sixth column. If we ever expand our interest in _mim2gene_medgen_ beyond the GPI genes, might have to worry about other values one can find in the sixth column:
```
$cut -f6 mim2gene_medgen | sort | uniq
-
Comment
nondisease
nondisease; QTL 2
nondisease; QTL 2; susceptibility
nondisease; nondisease; QTL 2
nondisease; question
nondisease; susceptibility
question
susceptibility
susceptibility; QTL 1
susceptibility; modifier
susceptibility; modifier; question
susceptibility; question
susceptibility; somatic
```
